constants.py

```diff
@@ -476,6 +485,87 @@
 kCameraServoIGain = 0.0
 kCameraServoDGain = 0.0
 
+# shooter parameters
+kTurretMotorId = 22
+kShootingMotorId = 23
+kHoodMotorId = 21
+
+kSimTurretMotorPort = 9
+kSimShootingMotorPort = 15
+kSimHoodMotorPort = 16
+
+kSimTurretMinimumLimitSwitchPort = 0
+kSimTurretMaximumLimitSwitchPort = 1
+kSimHoodMinimumSwitchPort = 2
+kSimHoodMaximumSwitchPort = 3
+
+kShootingPIDSlot = 0
+kShootingPGain = 0.7
+kShootingIGain = 0
+kShootingDGain = 0
+kShootingMotorInverted = True
+
+kShootingMappingFunction = lambda x: 16.41 * math.pow(math.e, 0.3662 * x) + 532
+
+kTurretPIDSlot = 0
+kTurretPGain = 0.06
+kTurretIGain = 0
+kTurretDGain = 0
+kTurretMotorInverted = False
+
+kTurretGearRatio = (1 / 5) * (24 / 200)
+
+kHoodPIDSlot = 0
+kHoodPGain = 0.15
+kHoodIGain = 0
+kHoodDGain = 0
+
+kHoodGearRatio = (1 / 5) * (-13 / 360)
```

Comment:
Gear ratios are generally not negative. If the output needs to be inverted that can be handled in the motor configuration.